test/hipcub/test_hipcub_device_radix_sort.hpp

```diff
@@ -613,9 +613,10 @@ inline void sort_keys_over_4g()
 
     hipDeviceProp_t dev_prop;
     HIP_CHECK(hipGetDeviceProperties(&dev_prop, device_id));
+    
     // Radix sort requires 2 buffers of `size`, so a minimum of 8 GB of vram for this test.
     // This is more than some cards provide.
-    if(dev_prop.totalGlobalMem < size * 2 * sizeof(key_type))
+    if(static_cast<size_t>(dev_prop.totalGlobalMem * 0.9) < size * 2 * sizeof(key_type))
```

Comment:
```suggestion
    size_t free_mem, total_mem;
    hipMemGetInfo(&free_mem, &total_mem);
    if(free_mem < size * 2 * sizeof(key_type))
```
Try this as well, please.